index.js

```diff
@@ -50,18 +54,34 @@ Cloneable.prototype.clone = function () {
   // for starting the flow
   this.removeListener('newListener', onData)
   var clone = new Clone(this)
-  this.on('newListener', onData)
+  if (this._hasListener) {
+    this.on('newListener', onData)
+  }
 
   return clone
 }
 
+Cloneable.prototype._destroy = function (err, cb) {
+  if (!err) {
+    this.push(null)
+    this.end()
```

Comment:
i'd get rid of the push(null) and end() calls. this makes error handling *super* tricky as the stream will look like it ended gracefully (unless that's what you are going for of course)